File: 230-二叉搜索树中第k小的元素/230_2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def kthSmallest(self, root: TreeNode, k: int) -> int:
        self.res,self.count = None, k
        def inorder(root):
            if not (root and self.count):
                return
            inorder(root.left)
            self.count -= 1
            if not self.count:
                self.res = root.val
            inorder(root.right)
        inorder(root)
        return self.res

File: 230-二叉搜索树中第k小的元素/230_2.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
  public int kthSmallest(TreeNode root, int k) {
      int index = 0;
      Stack<TreeNode> s = new Stack<>();
      if(root == null) return 0;
      while(root != null || !s.isEmpty()){
          while(root != null){
              s.push(root);
              root = root.left;
          }
          root = s.pop();
          index++;
          if(index == k){
              return root.val;
          }
          root = root.right;
      }
      return 0;
  }
}